torch_xla/experimental/scan.py

```diff
@@ -33,33 +73,65 @@ def scan(fn, init, xs):
         carry, y = fn(carry, xs[i])
         ys.append(y)
       return carry, torch.stack(ys, dim=0)
-  
+
   In the general case, `Carry`, `X`, and `Y` can be arbitrary PyTrees. This function
   will iterate through the leading dimension of every leaf element of `xs` simultaneously,
   and pass a slice of those elements to `fn` as another PyTree. This means you may
   scan over multiple tensors and produce multiple output tensors at once.
-  
-  Args:
 
-    fn: a Python callable that accepts two PyTrees of tensors: the carry object and the
-        slices of `xs` along its leading dimension. It should return two PyTrees: the carry
-        object and the slices of the output. The returned carry object will be passed to
-        the next invocation of `fn`.
+  Notes:
 
-    init: the initial carry object passed to the first invocation of `fn`.
+    `fn` must be AOTAutograd traceable. That requires PyTorch to understand the operations
+    within. For example if you invoke a custom kernel inside `fn`, you need to register the
+    custom kernel. See https://pytorch.org/tutorials/advanced/custom_ops_landing_page.html.
+
+  Args:
+    fn: A Python callable that accepts two PyTrees of tensors: the carry object and the
+      slices of `xs` along its leading dimension. It should return two PyTrees: the carry
+      object and the slices of the output. The returned carry object will be passed to
+      the next invocation of `fn`.
+
+    init: The initial carry object passed to the first invocation of `fn`.
     
-    xs: the input PyTree to scan over. If `xs` is a tensor, then `fn` will get slices along
-        the leading dimension (`xs[i]`). If `xs` is some other PyTree (e.g. tuple of
-        tensor), `fn` will get PyTrees of slices. In that case the leading dimension size
-        of the leaves in the PyTree must be the same.
+    xs: The input PyTree to scan over. If `xs` is a tensor, then `fn` will get slices along
+      the leading dimension (`xs[i]`). If `xs` is some other PyTree (e.g. tuple of
+      tensor), `fn` will get PyTrees of slices. In that case the leading dimension size
+      of the leaves in the PyTree must be the same.
+        
+    partition_fn: Since `scan` uses AOTAutograd to trace `fn`, you may override what
```

Comment:
if this is optional, you want to mark it as optional in the doc string

Reply:
done